manifests/params.pp

```diff
@@ -29,9 +29,18 @@
   if $::operatingsystem == 'Ubuntu' {
     if versioncmp($::lsbdistrelease, '8.04') < 1 {
       $init_style = 'debian'
-    } else {
+    } elsif versioncmp($::operatingsystemrelease, '15.04') < 0 {
       $init_style = 'upstart'
     }
+    else {
```

Comment:
I would prefer a "tell don't ask" approach here. What if we default to the stock init system of the operating systems we support, and if users want to force a different one, they can via the class param.

That is what the params class is for: providing sane defaults on a per-environment basis, but allowing users to override them. Because in theory the same thing could be happening on debian, or arch, etc.